### DISCUSSION TITLE
How to access Injector  in `ConstructorN`?

### DISCUSSION THREAD

**Discussion by zhangtianxiao** [activej/activej#223] [Q&A] [answered]
code like this
```java
Injector injector = Injector.of(
      ModuleBuilder.create()

        .bind(int.class).to(() -> 123)

        .bind(Example.class).to(()-> {
          // Variable 'injector' might not have been initialized
          Integer i = injector.getInstance(int.class);
          return new ExampleWare(i);
        })

        .bindOptionalDependency(String.class)
        .build());
```
Thanks!

**Comment by zhangtianxiao** [comment]
@eduard-vasinskyi

**Comment by eduard-vasinskyi** [comment, marked as answer]
Hi, @zhangtianxiao.

If you want to initialize `ExampleWare` with the provided `int` value, you do not need an Injector instance. All you need to do is request the provided `int` value when binding `Example.class`. To do this, you need to pass the desired classes or ActiveJ Inject [Key](https://github.com/activej/activej/blob/943d56f7f516f219b1846d4be346c22e6b49e6a8/core-inject/src/main/java/io/activej/inject/Key.java)s to the `ModuleBuilder0#to` method after the first `ConstructorN` argument.

Your binding for the `Example.class` would look like this:
```java
…
.bind(Example.class).to(ExampleWare::new, int.class)
…
```
 
If for some reason you still need to access the Injector instance, you can also request it when creating a binding. A binding for `Injector` is always present, so you do not have to bind it yourself.

Your code will look like this:
```java
...
.bind(Example.class).to(inj-> {
  Integer i = inj.getInstance(int.class);
  return new ExampleWare(i);
}, Injector.class)
...
```
